packages/react-native-web/src/exports/Image/index.js

```diff
@@ -94,61 +95,105 @@ function getFlatStyle(style, blurRadius, filterId) {
   return [flatStyle, resizeMode, _filter, tintColor];
 }
 
-function resolveAssetDimensions(source) {
-  if (typeof source === 'number') {
-    const { height, width } = getAssetByID(source);
-    return { height, width };
-  } else if (
-    source != null &&
-    !Array.isArray(source) &&
-    typeof source === 'object'
-  ) {
-    const { height, width } = source;
-    return { height, width };
-  }
+function resolveAssetDimensions(source: ImageSource) {
+  const { height, width } = source;
+  return { height, width };
 }
 
-function resolveAssetUri(source): ?string {
-  let uri = null;
+function resolveSource(source: ?Source): ImageSource {
+  let resolvedSource = { uri: '' };
+
   if (typeof source === 'number') {
-    // get the URI from the packager
-    const asset = getAssetByID(source);
-    if (asset == null) {
-      throw new Error(
-        `Image: asset with ID "${source}" could not be found. Please check the image source or packager.`
-      );
-    }
-    let scale = asset.scales[0];
-    if (asset.scales.length > 1) {
-      const preferredScale = PixelRatio.get();
-      // Get the scale which is closest to the preferred scale
-      scale = asset.scales.reduce((prev, curr) =>
-        Math.abs(curr - preferredScale) < Math.abs(prev - preferredScale)
-          ? curr
-          : prev
+    resolvedSource = resolveNumericSource(source);
+  } else if (typeof source === 'string') {
+    resolvedSource = resolveStringSource(source);
+  } else if (Array.isArray(source)) {
+    if (process.env.NODE_ENV !== 'production') {
+      console.warn(
+        'The <Image> component does not support multiple sources passed as array, falling back to the first source in the list',
+        { source }
       );
     }
-    const scaleSuffix = scale !== 1 ? `@${scale}x` : '';
-    uri = asset
-      ? `${asset.httpServerLocation}/${asset.name}${scaleSuffix}.${asset.type}`
-      : '';
-  } else if (typeof source === 'string') {
-    uri = source;
+
+    return resolveSource(source[0]);
```

Comment:
Added this Array handling case
Since arrays aren't covered we try to recover by using the first object in the array